File: KI305/Symanyshyn/Lab2/Logger.java
package KI305.Symanyshyn.Lab2;

import java.io.FileWriter;
import java.io.IOException;
import java.io.PrintWriter;

/**
 * Responsible for logging actions to a file.
 */
public class Logger {
    private PrintWriter logWriter;

    /**
     * Initializes the logger and opens the log file.
     */
    public Logger() {
        try {
            logWriter = new PrintWriter(new FileWriter("ClimberEquipment.log", true));
            logAction("Logger initialized.");
        } catch (IOException e) {
            System.out.println("Could not initialize logger: " + e.getMessage());
        }
    }

    /**
     * Logs an action to the log file.
     *
     * @param action The action to log.
     */
    public void logAction(String action) {
        if (logWriter != null) {
            logWriter.println(action);
        }
    }

    /**
     * Closes the logger and releases resources.
     */
    public void close() {
        if (logWriter != null) {
            logAction("Logger closed.");
            logWriter.close();
        }
    }
}

File: climbing_equipment/logger.py
class Logger:
    """
    Responsible for logging actions to a file.
    """
    def __init__(self, filename="ClimberEquipment.log"):
        self.filename = filename
        with open(self.filename, "a", encoding="utf-8") as file:
            file.write("Logger initialized.\n")

    def log_action(self, action):
        """
        Logs an action to the log file.
        :param action: The action to log.
        """
        with open(self.filename, "a", encoding="utf-8") as file:
            file.write(action + "\n")

    def close(self):
        """
        Closes the logger.
        """
        with open(self.filename, "a", encoding="utf-8") as file:
            file.write("Logger closed.\n")
